draft-ietf-dprive-dnsoquic.md

```diff
@@ -382,11 +385,27 @@ messages during a transaction. These include (but are not limited to)
 
 If a peer encounters such an error condition it is considered a fatal error. It
 SHOULD forcibly abort the connection using QUIC's CONNECTION_CLOSE mechanism,
-and use the DoQ error code DOQ_PROTOCOL_ERROR.
+and SHOULD use the DoQ error code DOQ_PROTOCOL_ERROR.
 
 It is noted that the restrictions on use of the above EDNS(0) options has
 implications for proxying message from TCP/DoT/DoH over DoQ.
 
+### Alternative error codes
+
+This specification suggests specific error codes {{transaction-cancellation}},
+{{transaction-errors}}, and {{protocol-errors}}. These error codes are meant
+to facilitates investigation of failures and other incidents. New error
+codes may be defined in future versions of DoQ, or registered as specified
+in {{iana-error-codes}}.
+
+Because new error codes can be defined without negotiation, use of an error
+code in an unexpected context or receipt of an unknown error code MUST be
+treated as equivalent to DOQ_NO_ERROR.
+
+Implementations that wish to test the support for the error code extension
+mechanism MAY use DOQ_ERROR_RESERVED, or any error code registered according
+to {{iana-error-codes}}.
```

Comment:
I don't think this needs to be so restrictive. If an endpoint wants to test something unknown, they can pick any legal value that is or is not registered.